api/capi/src/nnstreamer-capi-single.c

```diff
@@ -143,6 +145,80 @@ __setup_in_out_tensors (ml_single * single_h)
   }
 }
 
+/**
+ * @brief setup the destroy notify for the allocated output data.
+ * @note this stores the data entry in the single list.
+ * @note this has not overhead if the allocation of output is not performed by
+ * the framework but by tensor filter element.
+ */
+static inline void
```

Comment:
Q: Is there any reason to make this function inline?

Reply:
Yes, this is in the critical path of the `invoke` function.
Also, this is a very small function that has been moved out of `__process_output` for readability.

Reply:
If this is in the critical path as you mentioned, IMO, adding the ```inline``` keyword to all those functions related to __process_output does not guarantee the performance improvement.  It is possible to make the body in the while loop of invoke_thread () be large.

Reply:
As you already know, `__proces_output` and `__invoke` are used in both `invoke_thread` and `ml_single_invoke`.
And I don't think copying the same code in both functions inside is a good choice.

Reply:
> As you already know, `__proces_output` and `__invoke` are used in both `invoke_thread` and `ml_single_invoke`.
> And I don't think copying the same code in both functions inside is a good choice.

I am not saying that this function is unnecessary. I am concerning the ```inline`` keyword only.

Reply:
I meant that given `__process_output` is inline, I didn't want to introduce a non-inline function inside it (which would be against the inline attribute of `__process_output`). Which one do you recommend -
1. don't make `set_destroy_notify`, and let that code directly be in `__process_output`
2. have `set_destroy_notify`
3. have `set_destroy_notify` with inline attribute
4. something else

Reply:
2

Reply:
done.